src/components/molecules/Schedule/Schedule.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useCallback, useMemo, useState } from "react";
+import classNames from "classnames";
+import {
+  eachHourOfInterval,
+  endOfDay,
+  format,
+  getHours,
+  getUnixTime,
+  setHours,
+} from "date-fns";
+import { useCss } from "react-use";
+
+import {
+  SCHEDULE_CURRENT_TIMELINE_MS,
+  SCHEDULE_HOUR_COLUMN_WIDTH_PX,
+  SCHEDULE_MAX_START_HOUR,
+} from "settings";
+
+import { PersonalizedVenueEvent, LocatedEvents } from "types/venues";
+
+import { eventStartTime } from "utils/event";
+import { formatMeasurement } from "utils/formatMeasurement";
+
+import { useInterval } from "hooks/useInterval";
+
+import { ScheduleRoomEvents } from "components/molecules/ScheduleRoomEvents";
+
+import { calcStartPosition } from "./Schedule.utils";
+
+import "./Schedule.scss";
+
+export interface ScheduleProps {
+  locatedEvents: LocatedEvents[];
+  personalEvents: PersonalizedVenueEvent[];
+  scheduleDate: Date;
+  isToday: boolean;
+}
+
+export const Schedule: React.FC<ScheduleProps> = ({
+  locatedEvents,
+  personalEvents,
+  scheduleDate,
+  isToday,
+}) => {
+  const hasEvents = locatedEvents.length > 0;
+
+  const scheduleStartHour = useMemo(
+    () =>
+      Math.min(
+        ...locatedEvents.map(({ events }) =>
+          events.reduce(
+            (acc, event) => Math.min(acc, getHours(eventStartTime(event))),
+            SCHEDULE_MAX_START_HOUR
+          )
+        ),
+        SCHEDULE_MAX_START_HOUR
+      ),
+    [locatedEvents]
+  );
```

Comment:
@sunny-viktoryia ?

Reply:
> Seems to be finding the earliest start time across all of the events?

yes, the earliest start time across all the events that are happening at the selected day (they are filtered earlier).

>What's the purpose of SCHEDULE_MAX_START_HOUR here?
```js
// settings.ts
const SCHEDULE_MAX_START_HOUR = 16; // in the current feature version
```

16 is the latest hour at which should the schedule start. That is, if all events start after 10 pm, the schedule will still be displayed from 4pm so that the half-screen is not empty. I adjusted it based on my display size.
But! Daniel, for example, has a super-long screen, and he still has space on the right. Therefore, ideally, it would be to calculate the maximum number to fit the screen. This is something should be done I guess.
I've added a comment to settings
```js
// @debt probably would be better to adjust max hour based on user's display size
```

Reply:
Ah ok, that makes sense :)